lumigator/python/mzai/backend/backend/api/routes/jobs.py

```diff
@@ -103,20 +105,57 @@ def list_jobs(
     )
 
 
-@router.get("/{job_id}")
-def get_job(service: JobServiceDep, job_id: UUID) -> Job:
+@router.get("/{job_spec}")
+def get_job(
+    service: JobServiceDep,
+    job_spec: str,
+    skip: int = 0,
+    limit: int = 100,
+) -> Job | ListingResponse[Job]:
     """Attempts to retrieve merged job data from the Lumigator repository and Ray.
+    If a valid UUID is provided, a single record is returned. If a valid
+    job type (currently `inference` or `evaluation`) is provided, then a list
```

Comment:
I would split this functionality into two methods, one that handles UUIDs and another that handles list of jobs by type

Reply:
Basically https://github.com/mozilla-ai/lumigator/pull/576#discussion_r1908935742, right?

Reply:
Yep